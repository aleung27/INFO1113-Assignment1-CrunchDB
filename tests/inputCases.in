SeT f 4
Set 0 3 1 24
helP
snApShoT
LisT KeyS
List SNapsHOts
LisT enTries
GEt f
dEl f
PurgE f
pOP 0
DRoP 1
RollBaCk 1
CHeCKouT 99
reStore nonexistent.txt
MiN 0
MAx 0
SuM 0
LEn 0
ReV 0
uniQ 0
SorT 0
PuSh 0 7
AppEND 0 0
PIck 0 1
PlUCK 0 1
ArchiVE 99 cases.txt
DiFF 0 f
IntEr 0 f
UnioN 0 f
CARTProd 0 f
HelP
byE
