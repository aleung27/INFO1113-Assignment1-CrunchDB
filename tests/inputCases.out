> ok

> ok

> BYE   clear database and exit
HELP  display this help message

LIST KEYS       displays all keys in current state
LIST ENTRIES    displays all entries in current state
LIST SNAPSHOTS  displays all snapshots in the database

GET <key>    displays entry values
DEL <key>    deletes entry from current state
PURGE <key>  deletes entry from current state and snapshots

SET <key> <value ...>     sets entry values
PUSH <key> <value ...>    pushes values to the front
APPEND <key> <value ...>  appends values to the back

PICK <key> <index>   displays value at index
PLUCK <key> <index>  displays and removes value at index
POP <key>            displays and removes the front value

DROP <id>      deletes snapshot
ROLLBACK <id>  restores to snapshot and deletes newer snapshots
CHECKOUT <id>  replaces current state with a copy of snapshot
SNAPSHOT       saves the current state as a snapshot

ARCHIVE <id> <filename> saves snapshot to file
RESTORE <filename> loads snapshot from file

MIN <key>  displays minimum value
MAX <key>  displays maximum value
SUM <key>  displays sum of values
LEN <key>  displays number of values

REV <key>   reverses order of values
UNIQ <key>  removes repeated adjacent values
SORT <key>  sorts values in ascending order

DIFF <key> <key ...>   displays set difference of values in keys
INTER <key> <key ...>  displays set intersection of values in keys
UNION <key> <key ...>  displays set union of values in keys
CARTPROD <key> <key ...>  displays set union of values in keys

> saved as snapshot 1

> 0
f

> 1

> 0 [3 1 24]
f [4]

> [4]

> ok

> ok

> 3

> ok

> no such snapshot

> no such snapshot

> file not found

> 1

> 24

> 25

> 2

> ok

> ok

> ok

> ok

> ok

> 7

> 7

> no such snapshot

> no such key

> no such key

> no such key

> no such key

> BYE   clear database and exit
HELP  display this help message

LIST KEYS       displays all keys in current state
LIST ENTRIES    displays all entries in current state
LIST SNAPSHOTS  displays all snapshots in the database

GET <key>    displays entry values
DEL <key>    deletes entry from current state
PURGE <key>  deletes entry from current state and snapshots

SET <key> <value ...>     sets entry values
PUSH <key> <value ...>    pushes values to the front
APPEND <key> <value ...>  appends values to the back

PICK <key> <index>   displays value at index
PLUCK <key> <index>  displays and removes value at index
POP <key>            displays and removes the front value

DROP <id>      deletes snapshot
ROLLBACK <id>  restores to snapshot and deletes newer snapshots
CHECKOUT <id>  replaces current state with a copy of snapshot
SNAPSHOT       saves the current state as a snapshot

ARCHIVE <id> <filename> saves snapshot to file
RESTORE <filename> loads snapshot from file

MIN <key>  displays minimum value
MAX <key>  displays maximum value
SUM <key>  displays sum of values
LEN <key>  displays number of values

REV <key>   reverses order of values
UNIQ <key>  removes repeated adjacent values
SORT <key>  sorts values in ascending order

DIFF <key> <key ...>   displays set difference of values in keys
INTER <key> <key ...>  displays set intersection of values in keys
UNION <key> <key ...>  displays set union of values in keys
CARTPROD <key> <key ...>  displays set union of values in keys

> bye
